source/mqtt_agent.c

```diff
@@ -922,7 +922,11 @@ MQTTStatus_t MQTTAgent_CommandLoop( MQTTAgentContext_t * pMqttAgentContext )
     bool endLoop = false;
 
     /* The command queue should have been created before this task gets created. */
-    if( ( pMqttAgentContext == NULL ) || ( pMqttAgentContext->agentInterface.pMsgCtx == NULL ) )
+    if( ( pMqttAgentContext == NULL ) )
+    {
+        operationStatus = MQTTBadParameter;
+    }
+    else if( pMqttAgentContext->agentInterface.pMsgCtx == NULL )
```

Comment:
Revert this change, it's a MISRA violation to not have an `else` after an `else if` and it looks cleaner with just one `if`